argo/providers/vai/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"cloud.google.com/go/storage"
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	. "github.com/sky-uk/kfp-operator/providers/base"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+	"io"
+	"log"
+	"os"
+)
+
+func main() {
+	err := RunProviderApp[VertexAiProviderConfig](VAIProvider{})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+type VertexAiProviderConfig struct {
+	PipelineBucket string `yaml:"pipelineBucket,omitempty"`
+	Endpoint       string `yaml:"endpoint,omitempty"`
+}
+
+type VAIProvider struct {
+}
+
+func (vaip VAIProvider) client(providerConfig VertexAiProviderConfig, ctx context.Context) (*storage.Client, error) {
+	var client *storage.Client
+	var err error
+	if providerConfig.Endpoint != "" {
+		client, err = storage.NewClient(ctx, option.WithoutAuthentication(), option.WithEndpoint(providerConfig.Endpoint))
+	} else {
+		client, err = storage.NewClient(ctx)
+	}
+
+	return client, err
+}
+
+func (vaip VAIProvider) CreatePipeline(providerConfig VertexAiProviderConfig, pipelineConfig PipelineConfig, pipelineFile string, ctx context.Context) (string, error) {
+	client, err := vaip.client(providerConfig, ctx)
+
+	if err != nil {
+		return "", err
+	}
+
+	id, err := uuid.NewUUID()
+	if err != nil {
+		return "", err
+	}
+
+	writer := client.Bucket(providerConfig.PipelineBucket).Object(fmt.Sprintf("%s/", id)).NewWriter(ctx)
+	_, err = writer.Write([]byte("abc"))
```

Comment:
this should not be here

Reply:
in fact, the create workflow should not create and update and just create instead.

Reply:
> in fact, the create workflow should not create and update and just create instead.
This will be addressed in a separate PR